MediaBrowser.Controller/Entities/Folder.cs

```diff
@@ -842,25 +842,28 @@ private bool RequiresPostFiltering(InternalItemsQuery query)
 
         private static BaseItem[] SortItemsByRequest(InternalItemsQuery query, IReadOnlyList<BaseItem> items)
         {
-            var ids = query.ItemIds;
             int size = items.Count;
 
             // ids can potentially contain non-unique guids, but query result cannot,
             // so we include only first occurrence of each guid
             var positions = new Dictionary<Guid, int>(size);
             int index = 0;
-            for (int i = 0; i < ids.Length; i++)
+            foreach (var id in query.ItemIds)
             {
-                if (positions.TryAdd(ids[i], index))
+                if (positions.TryAdd(id, index))
                 {
                     index++;
                 }
             }
 
+            if (index > size)
+            {
+                size = index;
+            }
+
             var newItems = new BaseItem[size];
-            for (int i = 0; i < size; i++)
+            foreach (var item in items)
```

Comment:
You can't replace the for-loop with foreach like this. Either keep the for-loop or do a foreach over `positions`. Right now you've changed the behaviour to grab the _last_ occurrence of the duplicate instead of the first one.

Looping over items and using a hashset to keep track of already-added IDs might be faster, but you'd have to bench it.